internal/engine/eval/vulncheck/review.go

```diff
@@ -305,52 +256,93 @@ func (ra *reviewPrHandler) setStatus() {
 	ra.logger.Debug().Str("status", *ra.status).Msg("will set review status")
 }
 
-func (ra *reviewPrHandler) findPreviousReview(ctx context.Context) error {
-	reviews, err := ra.cli.ListReviews(ctx, ra.pr.RepoOwner, ra.pr.RepoName, int(ra.pr.Number), nil)
+func (ra *reviewPrHandler) findPreviousStatusComment(ctx context.Context) error {
+	comments, err := ra.cli.ListIssueComments(ctx, ra.pr.RepoOwner, ra.pr.RepoName, int(ra.pr.Number),
+		&github.IssueListCommentsOptions{
+			Sort:      github.String("updated"),
+			Direction: github.String("asc"),
+		})
 	if err != nil {
-		return fmt.Errorf("could not list reviews: %w", err)
+		return fmt.Errorf("could not list comments: %w", err)
 	}
 
-	ra.minderReview = nil
-	for _, r := range reviews {
-		if strings.HasPrefix(r.GetBody(), reviewBodyMagicComment) && r.GetState() != "DISMISSED" {
-			ra.minderReview = r
+	ra.minderStatusReport = nil
+	for _, comment := range comments {
+		isMinder := comment.GetUser().GetID() == ra.authorizedUser.GetID()
+		if isMinder && strings.HasPrefix(comment.GetBody(), statusBodyMagicCommentPrefix) {
+			ra.minderStatusReport = comment
 			break
 		}
 	}
 
 	return nil
 }
 
-func (ra *reviewPrHandler) submitReview(ctx context.Context) error {
-	body, err := createReviewBody(*ra.text)
+func (ra *reviewPrHandler) getMagicCommentInfo() magicCommentInfo {
+	if ra.minderStatusReport == nil {
+		return magicCommentInfo{}
+	}
+
+	mci, err := extractContentShaAndReviewID(ra.minderStatusReport.GetBody())
 	if err != nil {
-		return fmt.Errorf("could not create review body: %w", err)
+		ra.logger.Warn().Msg("could not extract content sha and review id from previous status comment")
+		// non-fatal error, we can still post the status report
+		// and worst case we will add a duplicate comment
+	}
+	return mci
+}
+
+func (ra *reviewPrHandler) submitReview(ctx context.Context, mci magicCommentInfo) (int64, error) {
```

Comment:
Future comment -- it would make sense to include a comment indicating what the returned int64 is.  In this case, it appears to be the issue / PR number.